### DISCUSSION TITLE
How to use hoppscotch desktop app in self-host environment

### DISCUSSION THREAD

**Discussion by simpleway** [hoppscotch/hoppscotch#3801] [Q&A] [answered]
Is there any instructions how to configure hoppscotch desktop app to use the self-hosted backend?

Thanks!

**Comment by giosueDelgado** [comment]
Interested!

**Comment by mrvincenzo** [comment]
Also very interested!

**Comment by preader5705** [comment]
Can the desktop app be used standalone or is a backend server required? I just downloaded the Windows desktop app, but when I run it, it just immediately terminates without any error message.

**Reply by preader5705** [reply]
Ignore my message. It looks like this issue was already reported: https://github.com/hoppscotch/hoppscotch/issues/3524

**Comment by ali-shokoohi** [comment]
Very very interested!

**Comment by SanskritiHarmukh** [comment, marked as answer]
Hey everyone, you can track the progress of adding self-hosted instance support to the desktop app [here](https://github.com/hoppscotch/hoppscotch/issues/3523).

**Comment by muellerst-hg** [comment]
The 2025.2.0 [release notes](https://hoppscotch.com/blog/hoppscotch-v2025-2-0) announced that Desktop App now supports self-hosted deployments. Is there any setup required on server side? I tried to add my instance in the desktop app using the drop down in the top left side, but it fails with "Verification error: Key validation failed: HTTP request failed: error decoding response body"
Self hosted instance and desktop app running on 2025.2.0

**Reply by liyasthomas** [reply]
I suspect this might be the problem: https://github.com/hoppscotch/hoppscotch/issues/3523#issuecomment-2697487626

**Reply by muellerst-hg** [reply]
Thanks for the pointer, but I think that's not related in my setup, which uses three independent containers. The containers are running the following docker images:
- `hoppscotch/hoppscotch-frontend:2025.2.0`
- `hoppscotch/hoppscotch-backend:2025.2.0`
- `hoppscotch/hoppscotch-admin:2025.2.0`

I compared these three images and the AIO image and as the result, AIO contains `/usr/local/bin/webapp-server`, but none of the other three images ships this file.

**Reply by muellerst-hg** [reply]
this particular issue is fixed with hoppscotch/hoppscotch-frontend:2025.2.2 which now contains `/usr/local/bin/webapp-server`
